### ISSUE TITLE
JsonFormat not working with introspection

### ISSUE THREAD

**Issue by morki** [micronaut-projects/micronaut-core#4418]
When using `@JsonFormat` annotation, everything works fine if introspection is not used. 

```kotlin
data class Test(
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    val test: LocalDateTime,
)
```

However, when annotated with '@Introspected`, error is thrown during deserialization.

```kotlin
@Introspected
data class Test(
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    val test: LocalDateTime,
)
```

```
Cannot deserialize value of type `java.time.LocalDateTime` from String \"2020-08-25 08:53:22\": Failed to deserialize java.time.LocalDateTime: (java.time.format.DateTimeParseException) Text '2020-08-25 08:53:22' could not be parsed at index 10
```

**Comment by prayadav** [comment]
@morki  Can you verify that micronaut annotation processor is enabled? Works for me in Java

**Comment by graemerocher** [comment]
Try `@field:JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")`

**Comment by morki** [comment]
Thank you guys, I tried that but it really does not work. Reproduction repo here: https://github.com/morki/micronaut-jsonformat-bug

By default (introspection enabled) It throws exception, but if you uncomment `bean-introspection-module: false` in `application.yml` it works fine.

**Comment by Alex100** [comment]
Ran into the same bug. As soon as ``@Introspected`` is removed, everything works fine. The ``@field:JsonFormat`` does not fix this problem.

**Comment by andersonzup** [comment]
`@field:NotNull  @field:JsonFormat(pattern = "yyyy-MM-dd", shape = JsonFormat.Shape.ARRAY) val publicacao: LocalDate`

This way it works, but it returns an array

**Comment by morki** [comment]
Fixed in #6546

**Comment by yawkat** [comment]
Thanks for testing it again.
